<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="one_arm">

<material name="gray">
  <color rgba="0.8 0.8 0.8 1"/>
</material>
<material name="gray-red">
  <color rgba="1.0 0.8 0.8 2.0"/>
</material>
<material name="gray-green">
  <color rgba="0.8 1.0 0.8 2.0"/>
</material>
<material name="gray-blue">
  <color rgba="0.8 0.8 1.0 2.0"/>
</material>

<link name="world">
</link>
<link name="temp1">
</link>
<link name="temp2">
</link>
<link name="base_link">
</link>

<link name="link1">
  <visual>
    <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.15" length="0.10" />
      </geometry>
      <material name="gray-blue">
    </material>
  </visual>
</link>
<link name="link2">
  <visual>
    <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.1 0.1 0.1" />
      </geometry>
      <material name="gray-green">
    </material>
  </visual>
</link>

<joint name="robot_joint1" type="prismatic">
  <parent link="world"/>
  <child link="temp1"/>
  <origin xyz="0 0 0" rpy="0 0 0"/>
  <axis xyz="1 0 0"/>
  <limit effort="30" velocity="1.0" lower="-2.0" upper="2.0" />
</joint>
<joint name="robot_joint2" type="prismatic">
  <parent link="temp1"/>
  <child link="temp2"/>
  <origin xyz="0 0 0" rpy="0 0 1.571"/>
  <axis xyz="1 0 0"/>
  <limit effort="30" velocity="1.0" lower="-2.0" upper="2.0" />
</joint>
<joint name="robot_joint3" type="revolute">
  <parent link="temp2"/>
  <child link="base_link"/>
  <origin xyz="0 0 0" rpy="0 0 -1.571"/>
  <axis xyz="0 0 1"/>
  <limit effort="30" velocity="1.0" lower="-2.0" upper="2.0" />
</joint>
<joint name="joint1" type="fixed">
  <parent link="base_link"/>
  <child link="link1"/>
  <origin xyz="0 0 0.05" rpy="0 0 0"/>
</joint>
<joint name="joint2" type="fixed">
  <parent link="base_link"/>
  <child link="link2"/>
  <origin xyz="0.08 0 0.1" rpy="0 0 0"/>
</joint>

</robot>
